### PR TITLE
Update OpenCensus Stackdriver exporter version

### PR DESCRIPTION
**Pull request by yanweiguo**
This makes sending metrics to Stackdriver using `knative_revision` resource type actually work. Including the following changes:

1. Update OpenCensus Stackdriver exporter version to support built-in metrics.
1. Decide which resource type to use based on full metric type instead of metric prefix. For none built-in metrics of `knative_revision`, send them to Stackdriver using `global` resource type with customer domain.
1. Split Prometheus exporter and Stackdriver exporter to their own files.

### CODE REVIEW

File: metrics/prometheus_exporter_test.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2018 The Knative Authors.
```
Comment (mattmoor-sockpuppet) [inline metrics/prometheus_exporter_test.go]:
```suggestion
Copyright 2019 The Knative Authors.
```

File: metrics/stackdriver_exporter_test.go [hunk 2]
```diff
@@ -0,0 +1,178 @@
+/*
+Copyright 2018 The Knative Authors.
```
Comment (mattmoor-sockpuppet) [inline metrics/stackdriver_exporter_test.go]:
```suggestion
Copyright 2019 The Knative Authors.
```

File: metrics/prometheus_exporter_test.go (hunk 1 above)
Comment (mattmoor-sockpuppet) [inline metrics/prometheus_exporter_test.go]:
```suggestion
Copyright 2019 The Knative Authors.
```

File: metrics/stackdriver_exporter_test.go (hunk 2 above)
Comment (mattmoor-sockpuppet) [inline metrics/stackdriver_exporter_test.go]:
```suggestion
Copyright 2019 The Knative Authors.
```

File: metrics/prometheus_exporter_test.go (hunk 1 above)
Comment (mattmoor-sockpuppet) [inline metrics/prometheus_exporter_test.go]:
```suggestion
Copyright 2019 The Knative Authors.
```

File: metrics/stackdriver_exporter_test.go (hunk 2 above)
Comment (mattmoor-sockpuppet) [inline metrics/stackdriver_exporter_test.go]:
```suggestion
Copyright 2019 The Knative Authors.
```

File: metrics/prometheus_exporter_test.go (hunk 1 above)
Comment (mattmoor-sockpuppet) [inline metrics/prometheus_exporter_test.go]:
```suggestion
Copyright 2019 The Knative Authors.
```

File: metrics/stackdriver_exporter_test.go (hunk 2 above)
Comment (mattmoor-sockpuppet) [inline metrics/stackdriver_exporter_test.go]:
```suggestion
Copyright 2019 The Knative Authors.
```

**Comment by yanweiguo** [comment]
/assign @vaikas-google 
/assign @mattmoor

I changed the .gitattributes to exclude test coverage check for `/metrics/gcp_metadata.go` since the code there calls real GCP metadata service. We probably do not want to do that in unit tests. PTAL.

File: metrics/config.go [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2018 The Knative Authors
+Copyright 2019 The Knative Authors
```
Comment (vaikas-google) [inline metrics/config.go]: We haven't been doing changes to copyright notices for existing files. Could you revert these, new files using 2019 is fine of course?

Comment (yanweiguo) [reply metrics/config.go]: Done. Thanks.

**Pull request merged**